File: tests/e2e/fixtures/enums/postgresql/java/src/main/java/com/example/db/com/example/db/models/Status.java
package com.example.db.models;

public enum Status {
    OPEN("open"),
    IN_PROGRESS("in_progress"),
    DONE("done"),
    CANCELLED("cancelled");

    private final String value;

    Status(String value) {
        this.value = value;
    }

    public String getValue() {
        return value;
    }

    public static Status fromValue(String value) {
        for (Status e : values()) {
            if (e.value.equals(value)) {
                return e;
            }
        }
        throw new IllegalArgumentException("Unknown Status: " + value);
    }
}

File: tests/e2e/fixtures/enums/postgresql/kotlin/src/main/kotlin/com/example/db/com/example/db/models/Status.kt
package com.example.db.models

enum class Status(val value: String) {
    OPEN("open"),
    IN_PROGRESS("in_progress"),
    DONE("done"),
    CANCELLED("cancelled");

    companion object {
        fun fromValue(value: String): Status =
            entries.first { it.value == value }
    }
}
